Quotient/room.cpp

```diff
@@ -623,7 +623,8 @@ QStringList Room::pinnedEventIds() const {
 QVector<const Quotient::RoomEvent*> Quotient::Room::pinnedEvents() const
 {
     QVector<const RoomEvent*> pinnedEvents;
-    for (const auto& evtId : pinnedEventIds())
+    const auto eventIds = pinnedEventIds();
+    for (const auto& evtId : eventIds)
```

Comment:
I'd be interested to look at the rationale being clazy logic in this case. `QStringList` is not a reference or pointer - why would it dangle?